app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -1012,8 +1012,8 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = true
     )
 
-    var fxaHasSyncedItems by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_fxa_has_synced_items),
+    var hasFxaAuthenticated by booleanPreference(
```

Comment:
Using `Settings`/`Preferences` to monitor the authentication state seems a little sus. Were there other approaches you tried that didn't pan out?

Reply:
The underlying issue is that auth is (relatively) slow, to the point where there will be noticeable pop-in of the recent synced tab if we wait for it to finish. To me that demonstrates that we need some kind of cache mechanism so that we can show the placeholders immediately on app startup. If you have any other suggestions, I'm definitely all ears. Is it specifically using `Settings`? Do we have another easy file abstraction that I've so far missed?

One alternative I can think of off the top of my head is to cache it in the `FxaAccountManager` itself, but that would require adding a storage layer there as well and would ultimately be the same type of implementation.

Reply:
(for the purposes of this PR, this can still land)

I was just curious if you looked into having this live in `AppState` instead. That way it could be observed in other places via `appState.observeAsComposableState` in case other areas of the app can easily consume this in Compose, if necessary in the future.

Reply:
I filed #24911 to move it into app state, but the more I think about it the more I wonder whether it's actually the right move. Here's some of my thoughts:

- `Store`s are generally meant to be an observable entity. Any component would need to observe changes to `hasFxaAuthenticated` can also observe changes to auth status directly from an `FxaAccountManager`. 
- `hasFxaAuthenticated` is really meant as a one-time read value on startup
- Even if we move it into the store, it will still need to be provided from settings or some other kind of cache mechanism, so I think we'd just be creating a level of indirection

Reply:
I think those concerns/points are fair

Reply:
I closed it for now, then. Let me know if want to do something different in the future, happy to revisit this again.